File: qqbot-gateway/src/main/java/com/mieai/qqbot/gateway/GatewayConnection.java
package com.mieai.qqbot.gateway;

import java.util.concurrent.CompletionStage;

/** One established Gateway WebSocket connection. */
public interface GatewayConnection {
    CompletionStage<Void> sendText(String payload);

    CompletionStage<Void> close();
}

File: qqbot-gateway/src/main/kotlin/com/mieai/qqbot/gateway/GatewayConnection.kt
package com.mieai.qqbot.gateway

import java.util.concurrent.CompletionStage

/** One established Gateway WebSocket connection. */
interface GatewayConnection {
    fun sendText(payload: String): CompletionStage<Void>

    fun close(): CompletionStage<Void>
}
